apinetlet/controllers/network_controller.go

```diff
@@ -148,27 +164,28 @@ func (r *NetworkReconciler) reconcile(ctx context.Context, log logr.Logger, netw
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	if network.Status.State == "" {
+		if err := r.patchNetworkPending(ctx, network); err != nil {
+			return ctrl.Result{}, fmt.Errorf("unable to patch network: %w", err)
+		}
+	}
```

Comment:
Why is the network patched to pending if the state is empty? If we e.g. already allocated an apinet `Network` for it, we should instead immediately patch it to allocated instead of pending and then allocated.

If this is for defaulting only, omit it for now. We can e.g. instead have admission / CRD defaulting do it (which I still wouldn't recommend / do now if we don't see the immediate need for it).

Also, usually the status of the resource is patched at the end of its reconciliation. Furthermore, patching it first and then again at the end of the reconciliation will produce patch conflicts, since the resource version might differ. The managed resource should only be patched *once* during reconciliation (this doesn't apply for dependent resources, e.g. if you create a secret for a resource, there you can have multiple modifications). This avoids the aforementioned conflicts and keeps the loop clean and responsive.